File: sparkcql-core/src/main/scala/gov/va/sparkcql/adapter/data/DataAdapter.scala
package gov.va.sparkcql.adapter.data

import scala.reflect.runtime.universe._
import org.apache.spark.sql.{SparkSession, DataFrame}
import javax.xml.namespace.QName

trait DataAdapter {

  val spark: SparkSession

  def isDataTypeDefined(dataType: QName): Boolean
  
  def acquire(dataType: QName): Option[DataFrame]
}

File: sparkcql-core/src/main/java/gov/va/sparkcql/adapter/data/DataAdapter.java
package gov.va.sparkcql.adapter.data;

import javax.xml.namespace.QName;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;

import gov.va.sparkcql.adapter.Adapter;

public interface DataAdapter extends Adapter {

    public abstract Boolean isDataTypeDefined(QName dataType);

    public abstract Dataset<Row> acquireData(QName dataType);
}